imap/jmap_contact.c

```diff
@@ -863,6 +863,34 @@ static int _add_othergroup_entries(struct jmap_req *req,
     return r;
 }
 
+typedef struct {
+    const char *key;
+    const char *prop;
+    char *type;
+    size_t size;
+    struct message_guid guid;
+} property_blob_t;
+
+static property_blob_t *property_blob_new(const char *key, const char *prop,
+                                          char *type, struct buf *data)
+{
+    property_blob_t *blob = xzmalloc(sizeof(property_blob_t));
+
+    blob->key = key;
+    blob->prop = prop;
+    blob->type = type;
+    blob->size = buf_len(data);
+    message_guid_generate(&blob->guid, buf_base(data), buf_len(data));
+
+    return blob;
+}
+
+static void property_blob_free(property_blob_t *blob)
```

Comment:
If the blob memory gets freed, then use a `**blobptr` argument and reset it to NULL. Otherwise there could be the risk of double frees later.